core/server/src/main/java/alluxio/worker/file/FileSystemWorker.java

```diff
@@ -128,13 +144,54 @@ public long createUfsFile(AlluxioURI ufsUri) throws FileAlreadyExistsException,
   }
 
   /**
+   * Opens a stream to the under file system file denoted by the temporary file id. This call
+   * will skip to the position specified in the file before returning the stream. The caller of
+   * this method is required to close the stream after they have finished operations on it.
+   *
+   * @param tempUfsFileId the worker specific temporary file id for the file in the under storage
+   * @param position the absolute position in the file to position the stream at before returning
+   * @return an input stream to the ufs file positioned at the given position
+   * @throws FileDoesNotExistException if the worker file id is invalid
+   * @throws IOException if an error occurs interacting with the under file system
+   */
+  public InputStream getUfsInputStream(long tempUfsFileId, long position)
+      throws FileDoesNotExistException, IOException {
+    return mUnderFileSystemManager.getInputStreamAtPosition(tempUfsFileId, position);
+  }
+
+  /**
+   * Returns the output stream to the under file system file denoted by the temporary file id.
+   * The stream should not be closed by the caller.
```

Comment:
It will be closed in complete/cancel file. The reason for the asymmetric behavior is because the under storage is not guaranteed to be able to support random writes, so we need to keep the stream alive. For reads, it is possible to keep it isolated per read and close the stream. I've added more details to the java doc.